File: snippets/dart_snippets/lib/lnurl_auth.dart
import 'package:dart_snippets/sdk_instance.dart';
import 'package:flutter_breez_liquid/flutter_breez_liquid.dart';

Future<void> lnurlAuth() async {
  // ANCHOR: lnurl-auth
  /// Endpoint can also be of the form:
  /// keyauth://domain.com/auth?key=val
  String lnurlAuthUrl =
      "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4excttvdankjm3lw3skw0tvdankjm3xdvcn6vtp8q6n2dfsx5mrjwtrxdjnqvtzv56rzcnyv3jrxv3sxqmkyenrvv6kve3exv6nqdtyv43nqcmzvdsnvdrzx33rsenxx5unqc3cxgeqgntfgu";

  InputType inputType = await parse(input: lnurlAuthUrl);
  if (inputType is InputType_LnUrlAuth) {
    LnUrlCallbackStatus result = await breezSDKLiquid.instance!.lnurlAuth(reqData: inputType.data);
    if (result is LnUrlCallbackStatus_Ok) {
      print("Successfully authenticated");
    } else {
      print("Failed to authenticate");
    }
  }
  // ANCHOR_END: lnurl-auth
}

File: snippets/wasm/lnurl_auth.ts
import { type BindingLiquidSdk } from '@breeztech/breez-sdk-liquid'

const exampleLnurlAuth = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: lnurl-auth
  // Endpoint can also be of the form:
  // keyauth://domain.com/auth?key=val
  const lnurlAuthUrl =
    'lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4excttvdankjm3lw3skw0tvdankjm3xdvcn6vtp8q6n2dfsx5mrjwtrxdjnqvtzv56rzcnyv3jrxv3sxqmkyenrvv6kve3exv6nqdtyv43nqcmzvdsnvdrzx33rsenxx5unqc3cxgeqgntfgu'

  const input = await sdk.parse(lnurlAuthUrl)
  if (input.type === 'lnUrlAuth') {
    const result = await sdk.lnurlAuth(input.data)
    if (result.type === 'ok') {
      console.log('Successfully authenticated')
    } else {
      console.log('Failed to authenticate')
    }
  }
  // ANCHOR_END: lnurl-auth
}
